monai/transforms/croppad/array.py

```diff
@@ -345,6 +346,52 @@ def __call__(self, img: np.ndarray):
         return cropper(img)
 
 
+class RandScaleCrop(RandSpatialCrop):
+    """
+    Subclass of :py:class:`monai.transforms.RandSpatialCrop`. Crop image with random size or specific size ROI.
+    It can crop at a random position as center or at the image center.
+    And allows to set the minimum and maximum scale of image size to limit the randomly generated ROI.
+
+    Args:
+        roi_scale: if `random_size` is True, it specifies the minimum crop size: `roi_scale * image spatial size`.
+            if `random_size` is False, it specifies the expected scale of image size to crop. e.g. [0.3, 0.4, 0.5].
+            If its components have non-positive values or very small scale value causes the crop size to be zero,
+            will use `1.0` instead, which means the input image size.
+        max_roi_size: if `random_size` is True and `roi_scale` specifies the min crop region size, `max_roi_scale`
+            can specify the max crop region size: `max_roi_scale * image spatial size`.
+            if None, defaults to the input image size. if its components have non-positive values or very small
+            scale value causes the crop size to be zero, will use `1.0` instead, which means the input image size.
+        random_center: crop at random position as center or the image center.
+        random_size: crop with random size or specified size ROI by `roi_scale * image spatial size`.
+            if True, the actual size is sampled from `rand(roi_scale, max_roi_scale) * image spatial size`.
+    """
+
+    def __init__(
+        self,
+        roi_scale: Union[Sequence[float], float],
+        max_roi_scale: Optional[Union[Sequence[float], float]] = None,
+        random_center: bool = True,
+        random_size: bool = True,
+    ) -> None:
+        super().__init__(roi_size=-1, max_roi_size=None, random_center=random_center, random_size=random_size)
+        self.roi_scale = roi_scale
+        self.max_roi_scale = max_roi_scale
+
+    def __call__(self, img: np.ndarray):
+        """
+        Apply the transform to `img`, assuming `img` is channel-first and
+        slicing doesn't apply to the channel dim.
+        """
+        img_size = img.shape[1:]
+        ndim = len(img_size)
+        self.roi_size = [int(r * s) for r, s in zip(ensure_tuple_rep(self.roi_scale, ndim), img_size)]
```

Comment:
need to double check this `int(r*s)` if img_size is small for example (192, 192, 5), and scale is 0.1

Reply:
Hi @wyli , I already added doc-string to indicate this case:
"If its components have non-positive values or very small scale value causes the crop size to be zero, will use `1.0` instead, which means the input image size."

Thanks.